bi_view_editor/models/models.py

```diff
@@ -0,0 +1,171 @@
+        - create/update the necessary columns/tables for fields,
+        - initialize new columns on existing rows,
+        - add the SQL constraints given on the model,
+        - add the indexes on indexed fields,
+
+        Also prepare post-init stuff to:
+        - add foreign key constraints,
+        - reflect models, fields, relations and constraints,
+        - mark fields to recompute on existing records.
+
+        Note: you should not override this method. Instead, you can modify
+        the model's database schema by overriding method :meth:`~.init`,
+        which is called right after this one.
+    """
+    # This monkey patch is meant to fix an error (probably
+    # introduced by https://github.com/odoo/odoo/pull/15412), while
+    # running an update all. The _auto_init() method invoked during
+    # an update all is the one of BaseModel, and not the one of Base.
+
+    # START OF patch
+    if _bi_view(self._name):
+        return
+    # END of patch
+
+    models.raise_on_invalid_object_name(self._name)
+
+    # This prevents anything called by this method (in particular default
+    # values) from prefetching a field for which the corresponding column
+    # has not been added in database yet!
+    self = self.with_context(prefetch_fields=False)
+
+    self.pool.post_init(self._reflect)
+
+    cr = self._cr
+    parent_store_compute = False
+    update_custom_fields = self._context.get('update_custom_fields', False)
+    must_create_table = not tools.table_exists(cr, self._table)
+
+    if self._auto:
+        if must_create_table:
+            tools.create_model_table(cr, self._table, self._description)
+
+        if self._parent_store:
+            if not tools.column_exists(cr, self._table, 'parent_left'):
+                self._create_parent_columns()
+                parent_store_compute = True
+
+        self._check_removed_columns(log=False)
+
+        # update the database schema for fields
+        columns = tools.table_columns(cr, self._table)
+
+        def recompute(field):
+            _logger.info("Storing computed values of %s", field)
+            recs = self.with_context(active_test=False).search([])
+            recs._recompute_todo(field)
+
+        for field in self._fields.values():
+            if not field.store:
+                continue
+
+            if field.manual and not update_custom_fields:
+                continue  # don't update custom fields
+
+            new = field.update_db(self, columns)
+            if new and field.compute:
+                self.pool.post_init(recompute, field)
+
+    if self._auto:
+        self._add_sql_constraints()
+
+    if must_create_table:
+        self._execute_sql()
+
+    if parent_store_compute:
+        self._parent_store_compute()
+
+
+models.BaseModel._auto_init = _auto_init
+
+
+class Base(models.AbstractModel):
+    _inherit = 'base'
+
+    @api.model
+    def _auto_init(self):
+        if not _bi_view(self._name):
```

Comment:
Why putting this here if already done the same in the monkey patch https://github.com/OCA/reporting-engine/pull/202/files#diff-c154b9950056aa3ae207dc2242d334c9R80?

Reply:
You're probably right, this part could be not necessary anymore. We should try what happens by removing it ...